aspnetcore/fundamentals/index.md

```diff
@@ -16,97 +16,121 @@ ms.custom: H1Hack27Feb2017
+The `Main` method uses `WebHostBuilder`, which follows the builder pattern to create a web application host. The builder has methods that define the web server (for example, `UseKestrel`) and the startup class (`UseStartup`). In the example above, the [Kestrel](xref:fundamentals/servers/kestrel) web server is used. Other web servers, such as [WebListener](xref:fundamentals/servers/weblistener), can be used by invoking the appropriate extension method. We'll show more about `UseStartup` in the next section.
+
+`WebHostBuilder` provides many optional methods, including `UseIISIntegration` for hosting in IIS and IIS Express, and `UseContentRoot` for specifying the root content directory. The `Build` and `Run` methods build the `IWebHost` object that will host the application and begin listening for HTTP requests.
+
+---
 
 ## Startup
 
-The `UseStartup` method on `WebHostBuilder` specifies the `Startup` class for your app.
+The `UseStartup` method on `WebHostBuilder` specifies the `Startup` class for your app:
+
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
+[!code-csharp[Main](../getting-started/sample/aspnetcoreapp/Program2x.cs?highlight=15&range=6-17)]
+
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
 
 [!code-csharp[Main](../getting-started/sample/aspnetcoreapp/Program.cs?highlight=7&range=6-17)]
 
-The `Startup` class is where you define the request handling pipeline and where any services needed by the app are configured. The `Startup` class must be public and contain the following methods:
+---
+
+The `Startup` class is where you define the request handling pipeline and where any services needed by the application are configured. The `Startup` class must be public and contain the following methods:
 
 ```csharp
 public class Startup
 {
+    // This method gets called by the runtime. Use this method to add services to the container.
     public void ConfigureServices(IServiceCollection services)
     {
     }
 
+    // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
     public void Configure(IApplicationBuilder app)
     {
     }
 }
 ```
 
-* `ConfigureServices` defines the services (see [Services](#services) below) used by your app (such as the ASP.NET Core MVC framework, Entity Framework Core, Identity, etc.)
+* `ConfigureServices` defines the services (see [Services](#services) below) used by your application (such as the ASP.NET Core MVC, Entity Framework Core, Identity, etc.).
```

Comment:
below can't easily be replaced here. But maybe 

`ConfigureServices` defines the services (see [Services](#services) below) used by ~your~ application (such as the ASP.NET Core MVC, Entity Framework Core, Identity, etc.).

or better

`ConfigureServices` defines the services (see [Services](#services) below) used by ~your application~  (such as the ASP.NET Core MVC, Entity Framework Core, Identity, etc.).